eng/scripts/bomhelpers.ps1

```diff
@@ -2,310 +2,312 @@
+  $latestPatchOrGAVersion = $sortedVersions | Where-Object { !($_.IsPrerelease) } | ForEach-Object { $_.RawVersion } | Select-Object -First 1
     
-    $mavenArtifactInfo = [MavenArtifactInfo]::new($ArtifactId, $latestPatchOrGAVersion, $latestReleasedVersion)
+  $mavenArtifactInfo = [MavenArtifactInfo]::new($ArtifactId, $latestPatchOrGAVersion, $latestReleasedVersion)
 
-    return $mavenArtifactInfo
+  return $mavenArtifactInfo
 }
 
 function GetPatchVersion([String]$ReleaseVersion) {
-    $ParsedSemver = [AzureEngSemanticVersion]::new($ReleaseVersion)
-    if (!$ParsedSemver) {
-        LogError "Unexpected release version:$($ReleaseVersion).Exiting..."
-        exit 1
-    }
+  $ParsedSemver = [AzureEngSemanticVersion]::new($ReleaseVersion)
+  if (!$ParsedSemver) {
+    LogError "Unexpected release version:$($ReleaseVersion).Exiting..."
+    exit 1
+  }
 
-    return "$($ParsedSemver.Major).$($ParsedSemver.Minor).$($ParsedSemver.Patch + 1)"
+  return "$($ParsedSemver.Major).$($ParsedSemver.Minor).$($ParsedSemver.Patch + 1)"
 }
 
 function GetRemoteName() {
-    $mainRemoteUrl = 'https://github.com/Azure/azure-sdk-for-java.git'
-    foreach ($rem in git remote show) {
-        $remoteUrl = git remote get-url $rem
-        if ($remoteUrl -eq $mainRemoteUrl) {
-            return $rem
-        }
+  $mainRemoteUrl = 'https://github.com/Azure/azure-sdk-for-java.git'
+  foreach ($rem in git remote show) {
+    $remoteUrl = git remote get-url $rem
+    if ($remoteUrl -eq $mainRemoteUrl) {
+      return $rem
     }
-    LogError "Could not compute the remote name."
-    return $null
+  }
+  LogError "Could not compute the remote name."
+  return $null
 }
 
 function GetPipelineName([string]$ArtifactId, [string]$ArtifactDirPath) {
-  $ciYmlFilePath = Join-Path ArtifactDirPath "ci.yml"
+  $ciYmlFilePath = Join-Path $ArtifactDirPath "ci.yml"
   if (Test-Path $ciYmlFilePath) {
     return  "java - " + $ArtifactId 
   }
   else {
-    $ciDirPath = Split-Path -Path $arInfo.ArtifactDirPath -Parent
+    $ciDirPath = Split-Path -Path $ArtifactDirPath -Parent
     $ciYmlFilePath = Join-Path $ciDirPath "ci.yml"
     if (Test-Path $ciYmlFilePath) {
-      return  "java - " + $arInfo.ServiceDirectoryName
+      $serviceDirectoryName = [System.IO.Path]::GetFileName($ciDirPath)
+      return  "java - " + $serviceDirectoryName
     }
   }
 }
 
-function TriggerPipeline($PatchInfos) {
-  $distinctPipelineNames = $PatchInfos | Select-Object {$_.PipelineName} | Get-Unique -AsString
+function TriggerPipeline($PatchInfos, $BranchName) {
+  $distinctPipelineNames = $PatchInfos | ForEach-Object { $_.PipelineName } | Get-Unique
   $distinctPipelineNames | ForEach-Object { 
-    Write-Output "Triggering pipeline {$_}"
-    $cmdoutput = az pipeline --name $_}
+    Write-Output "Triggering pipeline $_"
+    $cmdOutput = az pipelines run -o json --name ""$_"" --organization "https://dev.azure.com/azure-sdk" --project "internal" --branch ""$BranchName""
```

Comment:
Is the cmdOutput ever printed here, e.g. printing the link to the pipeline run? Also, I would handle errors here similar to my comment above re `$LASTEXITCODE`, especially if people aren't logged in, etc.